crates/uv-resolver/src/requires_python.rs

```diff
@@ -279,15 +287,76 @@ impl serde::Serialize for RequiresPython {
 impl<'de> serde::Deserialize<'de> for RequiresPython {
     fn deserialize<D: serde::Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
         let specifiers = VersionSpecifiers::deserialize(deserializer)?;
-        let bound = RequiresPythonBound(
+        let (lower_bound, upper_bound) =
             crate::pubgrub::PubGrubSpecifier::from_release_specifiers(&specifiers)
                 .map_err(serde::de::Error::custom)?
-                .iter()
-                .next()
-                .map(|(lower, _)| lower.clone())
-                .unwrap_or(Bound::Unbounded),
-        );
-        Ok(Self { specifiers, bound })
+                .bounding_range()
+                .map(|(lower_bound, upper_bound)| (lower_bound.cloned(), upper_bound.cloned()))
+                .unwrap_or((Bound::Unbounded, Bound::Unbounded));
+        Ok(Self {
+            specifiers,
+            range: RequiresPythonRange(
+                RequiresPythonBound(lower_bound),
+                RequiresPythonBound(upper_bound),
+            ),
+        })
+    }
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct RequiresPythonRange(RequiresPythonBound, RequiresPythonBound);
+
+impl RequiresPythonRange {
+    /// Initialize a [`RequiresPythonRange`] with the given bounds.
+    pub fn new(lower: RequiresPythonBound, upper: RequiresPythonBound) -> Self {
+        Self(lower, upper)
+    }
+
+    /// Returns the lower bound.
+    pub fn lower(&self) -> &RequiresPythonBound {
+        &self.0
+    }
+
+    /// Returns the upper bound.
+    pub fn upper(&self) -> &RequiresPythonBound {
+        &self.1
+    }
+}
+
+impl Default for RequiresPythonRange {
+    fn default() -> Self {
+        Self(
+            RequiresPythonBound(Bound::Unbounded),
+            RequiresPythonBound(Bound::Unbounded),
+        )
+    }
+}
+
+impl From<RequiresPythonRange> for Range<Version> {
+    fn from(value: RequiresPythonRange) -> Self {
+        match (value.0.as_ref(), value.1.as_ref()) {
```

Comment:
This should definitely be part of pubgrub, can you open an issue?

Reply:
https://github.com/pubgrub-rs/pubgrub/issues/253